vowpalwabbit/parse_args.cc

```diff
@@ -1325,10 +1325,20 @@ vw& parse_args(options_i& options, trace_message_t trace_listener, void* trace_c
 
     bool strict_parse = false;
     size_t ring_size;
+    int tmp;
     option_group_definition vw_args("VW options");
-    vw_args.add(make_option("ring_size", ring_size).default_value(256).help("size of example ring"))
+    vw_args.add(make_option("ring_size", tmp).default_value(256).help("size of example ring"))
         .add(make_option("strict_parse", strict_parse).help("throw on malformed examples"));
     options.add_and_parse(vw_args);
+    
+    if (tmp <= 0)
+    {
+      THROW("ring_size should be positive");
+    }
+    else
```

Comment:
If you remove the else then you don't need to declare ring_size separate to its initialization.